src/styles/main.scss

```diff
@@ -0,0 +1,78 @@
+$main-color: #eee;
+$hover-color: #ffbfcb;
+$font-family: Arial, Halvetica, san-serif;
+$week-days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+$columns: 7;
+$calendar-size: 100px;
+$number: 31;
+$days-gap: 1px;
+
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  width: ($calendar-size + $days-gap) * $columns;
+  flex-wrap: wrap;
+  display: flex;
+  gap: $days-gap;
+
+  &__day {
+    background-color: $main-color;
+    border: 1px solid black;
+    justify-content: center;
+    align-items: center;
+    display: flex;
+    width: $calendar-size;
+    height: $calendar-size;
+    position: relative;
+
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: '#{$number}';
+        font-family: $font-family;
+        font-weight: 500;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: $hover-color;
+      transition-duration: 0.5s;
```

Comment:
The transition property should be applied to the element itself, not on the hover state, to ensure the transition effect is smooth both when the mouse enters and leaves the element.